lib/Models/SdmxJson/SdmxJsonDataflowStratum.ts

```diff
@@ -269,55 +270,62 @@ export class SdmxJsonDataflowStratum extends LoadableStratum(
             ? Array.from(allowedOptionIdsSet)
             : [];
 
-          // Get codes by merging allowedOptionIds with codelist
-          let filteredCodesList =
-            (allowedOptionIds.length > 0
-              ? codelist?.codes?.filter(
-                  code =>
-                    allowedOptionIds && allowedOptionIds.includes(code.id!)
-                )
-              : // If no allowedOptions were found -> return all codes
-                codelist?.codes) ?? [];
-
-          // Create options object
-          // If modelOverride `options` has been defined -> use it
-          // Other wise use filteredCodesList
-          const overrideOptions = modelOverride?.options;
-          const options =
-            isDefined(overrideOptions) && overrideOptions.length > 0
-              ? overrideOptions.map(option => {
-                  return { id: option.id, name: option.name, value: undefined };
-                })
-              : filteredCodesList.map(code => {
-                  return { id: code.id!, name: code.name, value: undefined };
-                });
-
-          if (isDefined(options) && options.length > 0) {
-            // Use first option as default if no other default is provided
-            let selectedId: string | undefined = modelOverride?.allowUndefined
-              ? undefined
-              : options[0].id;
-
-            // Override selectedId if it a valid option
-            const selectedIdOverride = modelOverride?.selectedId;
-
-            if (
-              isDefined(selectedIdOverride) &&
-              options.find(option => option.id === selectedIdOverride)
-            ) {
-              selectedId = selectedIdOverride;
-            }
+          let options: StratumFromTraits<DimensionOptionTraits>[] = [];
+
+          // Only create options if less then 1000 values
+          if (allowedOptionIds.length < 1000) {
+            // Get codes by merging allowedOptionIds with codelist
+            let filteredCodesList =
+              (allowedOptionIds.length > 0
+                ? codelist?.codes?.filter(
+                    code =>
+                      allowedOptionIds && allowedOptionIds.includes(code.id!)
```

Comment:
i know this is outside the scope of this review, but we don't need to convert allowedOptionsIds back into an array, we could instead use `allowedOptionsIdsSet.has` which is faster than the `includes` method on arrays. Also if we make allowedOptionsIdsSet an empty set if constraints is not an array, then we save ourselves a few null checks.

Reply:
Yep that is much nicer!